src/StripeTests/Infrastructure/SerializationTest.cs

```diff
@@ -38,5 +37,17 @@ public void HandleNull()
             var reloaded = JsonConvert.DeserializeObject<TestObjectDateTime>(JsonConvert.SerializeObject(obj));
             Assert.Null(reloaded.Date);
         }
+
+        [Fact]
+        public void Serialize()
+        {
+            var json = GetResourceAsString("api_fixtures.events.customer_updated.json");
+            var evt = JsonConvert.DeserializeObject<Event>(json);
+            var serialized = JsonConvert.SerializeObject(evt, Formatting.Indented);
+            var reserialized = JsonConvert.SerializeObject(
+                JsonConvert.DeserializeObject<Event>(serialized),
+                Formatting.Indented);
+            Assert.Equal(serialized, reserialized);
+        }
```

Comment:
Can we try with something other than Event? Like I worry that both serialization results would have `data` null and we would not see it.

Also, serializing twice only proves a subset of the case right? Let's say I have a JSON and deserialize to Account. If some properties are not supported then they would never be deserialized, so when I serialize the first time the json has only a subset of what was originally in the fixture. When I deserialize again, I end up with what I had the first time and not what was in the original json.

So I know that we can't reliably use a fixture from stripe-mock since every time a new property would be added to the resource that unrelated would break and force you to update the resource. But we could use a local fixture that should never be broken.

It still has the problem of comparing 2 JSON strings that are formatted differently, but I think it'd give us more confidence.

Reply:
It's not really possible to retrieve the input JSON once it's been deserialized into a model class, save for really simple classes and/or specifically formatted input JSON files that don't really match actual API payloads.

For instance, all deletable resources have a `bool? Deleted` attribute. When serializing, this will add `"deleted": null` to the JSON string, which wouldn't be present in an actual API payload.

Also, serialization of expandable fields is basically broken right now (and I think has always been).

We can (and probably should) improve serialization by fixing all those special cases, but this is out of scope for this PR.